actix-router/src/url.rs

```diff
@@ -170,11 +170,7 @@ impl Quoter {
             idx += 1;
         }
 
-        cloned.map(|data| {
-            // SAFETY: we get data from http::Uri, which does UTF-8 checks already
-            // this code only decodes valid pct encoded values
-            unsafe { String::from_utf8_unchecked(data) }
-        })
+        cloned.map(|data| String::from_utf8_lossy(&data).into_owned())
```

Comment:
The latter should keep the original intention as possible but I don't have a strong opinion. Can it be expensive in a usual case?

Reply:
I don;t stron opinion either, but the code tries hard to avoid allocations. For example , it never allocates a new string when the input is not percent-encoded.

Reply:
Okay, so let's go ahead with the current and we can revisit here whenever it becomes a hot point.